native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -284,8 +285,22 @@ private static Object convertHeterogeneousSequence(String attributePrefix, boole
         BMap<BString, Object> mapJson = newJsonMap();
         for (BXml bxml : sequence) {
             if (isCommentOrPi(bxml)) {
+                //
```

Comment:
why there is no logic here

Reply:
If XML has comments or PI, we have to skip those when converting to JSON or record. So we are using this to skip those.

Reply:
Updated this by calling `continue`